ugoku-kun/main.js

```diff
@@ -152,20 +155,37 @@ mqttClient.on("connect", function () {
   console.log("Connected to MQTT broker");
   // topicをsubscribe
   mqttClient.subscribe(subscribeTopics, function (err) {
-      if (err) {
-          console.error("Error while subscribing:", err);
-      } else {
-          console.log("Subscribed to topic:", subscribeTopics);
-      }
+    if (err) {
+      console.error("Error while subscribing:", err);
+    } else {
+      console.log("Subscribed to topic:", subscribeTopics);
+    }
   });
 });
 
+let lastMessageTimestamp = Date.now();
+const lastMessageTime = document.getElementById("time-stamp");
+
 // mqttClientがメッセージを受け取った時にfireするコールバック
 mqttClient.on('message', function (topic, message) {
-  if(topic === voltageTopic) handleVoltageMessage(message);
-  if(topic === gamepadCmdVelTopic) handleGamepadCmdVelMessage(message);
+  console.log("Receiving a message!")
+  lastMessageTimestamp = Date.now();
+  lastMessageTime.innerText = lastMessageTimestamp
+  if (topic === voltageTopic) handleVoltageMessage(message);
+  if (topic === gamepadCmdVelTopic) handleGamepadCmdVelMessage(message);
 });
 
+//5秒以上、messageを受信しなかった時の処理
+setInterval(function () {
+  const currentTime = Date.now();
+  const pastTime = currentTime - lastMessageTimestamp;
+  if (pastTime >= 5000) {
+    console.log("offline!!!!!")
+    onlineGreenColor.style.backgroundColor = "lightgray";
```

Comment:
It is good to change  `onlineGreenColor` variable name to a more understandable name. 
for example: `onlineStatusIcon`